client.go

```diff
@@ -125,3 +133,15 @@ func (s *dateSchedule) Next(now time.Time) time.Time {
 
 	return d
 }
+
+//Condition modelates a group of conditions
+type Condition struct {
+	IsValidChannel ValidChannel
+	IsValidBot     ValidBot
+}
+
+//ValidChannel returns true if the passed Channel and msg are valid
+type ValidChannel func(channel Channel, msg interface{}) bool
```

Comment:
why is the msg an `interface{}`? It either is a `flamingo.Message` or a `flamingo.OutgoingMessage`, right?

Also, why do you need this functionality?